modules/sdk-coin-apt/src/apt.ts

```diff
@@ -76,8 +118,43 @@ export class Apt extends BaseCoin {
     return true;
   }
 
-  parseTransaction(params: ParseTransactionOptions): Promise<ParsedTransaction> {
-    throw new Error('Method not implemented.');
+  async parseTransaction(params: AptParseTransactionOptions): Promise<ParsedTransaction> {
+    const transactionExplanation = await this.explainTransaction({ txHex: params.txHex });
+    if (!transactionExplanation) {
+      throw new Error('Invalid transaction');
+    }
+    return {
+      inputs: [
+        {
+          address: transactionExplanation.sender,
+          amount: transactionExplanation.outputAmount,
+        },
+      ],
+      outputs: [
+        {
+          address: transactionExplanation.outputs[0].address,
+          amount: transactionExplanation.outputs[0].amount,
+        },
+      ],
+    };
+  }
+
+  /**
+   * Explain a Apt transaction
+   * @param params
+   */
+  async explainTransaction(params: ExplainTransactionOptions): Promise<AptTransactionExplanation> {
+    const factory = this.getBuilder();
+    let rebuiltTransaction: BaseTransaction;
+
+    try {
+      const transactionBuilder = factory.from(params.txHex);
+      rebuiltTransaction = await transactionBuilder.build();
+    } catch {
+      throw new Error('Invalid transaction');
+    }
+
+    return rebuiltTransaction.explainTransaction();
   }
 
   generateKeyPair(seed?: Buffer): KeyPair {
```

Comment:
this method doesn't have unit tests. Refer `modules/sdk-coin-hbar/test/unit/hbar.ts`.